daemon-tests/tests/liquidation.rs

```diff
@@ -0,0 +1,161 @@
+use daemon::projection::CfdState;
+use daemon_tests::confirm;
+use daemon_tests::expire;
+use daemon_tests::flow::next_with;
+use daemon_tests::flow::one_cfd_with_state;
+use daemon_tests::maia::OliviaData;
+use daemon_tests::mock_oracle_announcements;
+use daemon_tests::open_cfd;
+use daemon_tests::simulate_attestation;
+use daemon_tests::start_both;
+use daemon_tests::wait_next_state;
+use daemon_tests::OpenCfdArgs;
+use model::Dlc;
+use model::Price;
+use otel_tests::otel_test;
+use rand::prelude::SliceRandom;
+use rand::thread_rng;
+use rust_decimal::Decimal;
+
+#[otel_test]
+async fn given_open_cfd_when_oracle_attests_long_liquidation_price_can_liquidate() {
+    let (mut maker, mut taker) = start_both().await;
+
+    let oracle_data = OliviaData::example_0();
+
+    // We set the initial price to a value much higher than that of the price that the mock oracle
+    // will attest to. This is so that the price attested to falls within the long liquidation
+    // interval of the payout curve (the very first interval)
+    let future_attestation_price = oracle_data.attested_price();
+    let initial_price =
+        Price::new(Decimal::from(future_attestation_price * 8)).expect("positive price");
+
+    let order_id = open_cfd(
+        &mut taker,
+        &mut maker,
+        OpenCfdArgs {
+            initial_price,
+            oracle_data: oracle_data.clone(),
+            ..Default::default()
+        },
+    )
+    .await;
+
+    assert!(
+        is_attestation_price_in_any_interval_of_all_liquidation_events(
+            &taker.latest_dlc(),
+            future_attestation_price
+        )
+    );
+
+    taker.system.commit(order_id).await.unwrap();
+    confirm!(commit transaction, order_id, maker, taker);
+    wait_next_state!(order_id, maker, taker, CfdState::OpenCommitted);
+
+    expire!(cet timelock, order_id, maker, taker);
+
+    // We should be able to use any liquidation event to close the CFD
+    let random_liquidation_event = *taker
+        .latest_dlc()
+        .liquidation_event_ids()
+        .choose(&mut thread_rng())
```

Comment:
Could we potentially check all or at least some deterministic values that represent sane boundaries? 
I would like to avoid randomness when possible in the tests to avoid flaky tests.

Reply:
Fair point. I think checking all of them is a no-go because it either would take too long (24 times longer and even more once we have more frequent liquidation events) or require hacky tricks like erasing/overwriting events from the test DB.

Perhaps the lesson here is that we should not try to capture everything with this actor test. I will default back to just checking the first liquidation event. And consider adding a unit test somewhere else that verifies that all liquidation events are correct.